### ISSUE TITLE
Unable to make the irobot create2 work

### ISSUE THREAD

**Issue by hashimK** [drzaiusx11/create-oi#8]
Hello Jeff,
Im using node.js in windows environment. my serial port is COM4. On running the exampl-wander.js file (after editing the seril port).Im getting this message below. And as soon as I stop the program (Cntrl-C in cmd) the create2 goes into spot mode. Any help would be greatly appreciated.
 
serial port opened successfully
no data received from create... attempting to connect (again)
no data received from create... attempting to connect (again)

**Comment by drzaiusx11** [comment]
init() takes an optional 'version' argument for Create2 (all the instructions assume Create1 atm)

change:
`robot.init({ serialport: "/dev/tty.usbserial-A2001nf6" });` (NOTE: use your actual serial port, this is an example)

to

`robot.init({ serialport: "/dev/tty.usbserial-A2001nf6", version: 2 });`

let me know if that fixes it

**Comment by hashimK** [comment]
Hi Jeff,
That worked superb!!!..
Also, it would be great if you could let me know how to send mode commands to the robot via your node.js library. Mode commands like passive, clean and dock.

Many Thanks,
Hashim Kahily.

**Comment by drzaiusx11** [comment]
Hi Hashim,

You can use this undocumented API with the corresponding OI opcodes to change modes:
robot.setMode(131); // safe mode
robot.setMode(132); // full mode
robot.setMode(128); // passive mode

Technically setMode is just a passthrough to sending opcodes to the Create, so you could use it to trigger a demo as well if you send the correct opcodes in sequence (note: I haven't tested this, you may need 'waits' between commands):
robot.setMode(128); // passive mode
robot.setMode(136); // demo
robot.setMode(1); // wander and dock

**Comment by hashimK** [comment]
Hello Jeff,
Thanks for all the help. Your answers has really helped a lot!
One more trouble I ran into is how to read the battery percentage left in the create 2 robot.
Any ideas?

**Comment by drzaiusx11** [comment]
Currently there isn't a way to read the battery level unfortunately.  If you look up the specs in the create oi pdf on iRobot's website I'll definitely accept a pull request implementing that.

If you're not up for the implementation, if you create a feature request issue I'll put it on my TODO list

**Comment by hashimK** [comment]
Hello Jeff,
I got the way of knowing the creator's battery percentage. In your main.js I replaced the packet number responsible for distance and angle with 25 and 26 numbers. That way and with a little math I can get the battery percentage left on the robot. But that also means I cannot get the distance and angle values. That's fine for me. For others, this comment can be helpful if they want to know the battery percentage.
Thanks Anyways.
